lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -1540,12 +1607,398 @@ void amf_app_handle_initial_context_setup_rsp(
+    return M5G_AS_FAILURE;
+  }
+
+  // Message construction for PDU session modification
+  msg.security_protected.plain.amf.header.extended_protocol_discriminator =
+      M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.security_protected.plain.amf.header.message_type =
+      M5GMessageType::PDU_SESSION_MODIFICATION_COMMAND;
+  msg.header.security_header_type =
+      SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_CYPHERED;
+  msg.header.extended_protocol_discriminator = M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.header.sequence_number =
+      ue_context->amf_context._security.dl_count.seq_num;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd
+      .extended_protocol_discriminator.extended_proto_discriminator =
+      M5G_SESSION_MANAGEMENT_MESSAGES;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd.pdu_session_identity
+      .pdu_session_id = pdu_sess_mod_req->pdu_session_id;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd.pti.pti =
+      smf_ctx->smf_proc_data.pti;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd.message_type.msg_type =
+      static_cast<uint8_t>(M5GMessageType::PDU_SESSION_MODIFICATION_COMMAND);
+
+  for (int i = 0; i < smf_ctx->qos_flow_list.maxNumOfQosFlows; i++) {
+    if (smf_ctx->qos_flow_list.item[i]
+            .qos_flow_req_item.ul_tft.tftoperationcode) {
+      QOSRulesMsg qosRuleMsg;
+      qosRuleMsg.length = 0;
+
+      QOSRule qos_rule;
+      qosRuleMsg.iei = PDU_SESSION_AUTH_QOS_RULES_IE_TYPE;
+      qos_rule.len = 3;
+      qos_rule.rule_oper_code = smf_ctx->qos_flow_list.item[i]
+                                    .qos_flow_req_item.ul_tft.tftoperationcode;
+      if (PDU_SESSION_DEFAULT_QFI ==
+          smf_ctx->qos_flow_list.item[i]
+              .qos_flow_req_item.qos_flow_identifier) {
+        qos_rule.dqr_bit = 0x1;
+        qos_rule.qos_rule_id = 1;
+      } else {
+        qos_rule.dqr_bit = 0x0;
+        qos_rule.qos_rule_id = 2;
+      }
+      qos_rule.no_of_pkt_filters =
+          (0x0F & smf_ctx->qos_flow_list.item[i]
+                      .qos_flow_req_item.ul_tft.numberofpacketfilters);
+      qos_rule.qos_rule_precedence = 0xff;
+      qos_rule.spare = 0x0;
+      qos_rule.segregation = 0x0;
+      qos_rule.qfi =
+          smf_ctx->qos_flow_list.item[i].qos_flow_req_item.qos_flow_identifier;
+
+      // Add or Modify QOS Flow
+      if (smf_ctx->qos_flow_list.item[i].qos_flow_req_item.qos_flow_action ==
+              policy_action_add ||
+          smf_ctx->qos_flow_list.item[i].qos_flow_req_item.qos_flow_action ==
+              policy_action_del) {
+        if (smf_ctx->qos_flow_list.item[i]
+                .qos_flow_req_item.ul_tft.tftoperationcode ==
+            TRAFFIC_FLOW_TEMPLATE_OPCODE_CREATE_NEW_TFT) {
+          amf_app_fill_create_new_tft(
+              smf_ctx->qos_flow_list.item[i]
+                  .qos_flow_req_item.ul_tft.packetfilterlist.createnewtft,
+              &qos_rule);
+        } else if (smf_ctx->qos_flow_list.item[i]
```

Comment:
Create a local pointer that takes the values from smf_context for UL TFT. Later this pointer can be used.